### ISSUE TITLE
The restore script creates a folder instead of directly copying the files.

### ISSUE THREAD

**Issue by RedXXIII** [Intrinsically-Sublime/UT-Backup-Restore-Scripts#1]
Hello! I suppose my title is not very clear, and I am sorry for that. I'll try to explain it. But first I would like to say that I am not a dev neither I am very familiar with shell. So I am very thankful for your work! I hope that I could help in some way here.

Let's assume that I want to back up and restore my “Downloads” folder. If I use the “UT_Backup_Tool.sh” script, it will successfully create a “Downloads” folder in the “/Backup/UT_backup_[date]” folder. This folder contains the files and subfolders that I want to save. That is perfect.

If I use the “UT_Restore_Tool.sh” script to restore the “Downloads” folder, it will create a “Downloads” folder (containing all the files and subfolders to restore) inside the real “Downloads” folder in my device. That is not very convenient.

I tried directly with a push command, I can obtain the same result if a try this:
adb push ~/PC_path/Downloads /home/phablet/Downloads

To have the expected behaviour, I have to use this instead:
adb push ~/PC_path/Downloads /home/phablet

So, maybe, we should modify the script (line 52 and 69) from this
out_folder="/home/phablet/$(basename "$folder")/"
to this
out_folder="/home/phablet/"

What do you think? Is it better or is it dangerous?

**Comment by RedXXIII** [comment]
I just tried, but all the prompts are becoming “Would you like to restore /home/phablet/” which is not good. So lines 52 and 69 should not be modified.

Maybe the modification should be on lines 58 and 75. That is from
adb push "$folder" "$out_folder"
to
adb push "$folder" /home/phablet/
?

**Comment by RedXXIII** [comment]
I tried and that seems to work! Waiting for your opinion on that. =)

**Comment by Intrinsically-Sublime** [comment]
Sorry for the late reply. I do not currently have a phone with working adb to test this but I think the issue is actually just the trailing slash on line 52 needs to be removed. The slash tells it to put the source inside the directory instead of replacing it.
```
out_folder="/home/phablet/$(basename "$folder")/"
```
to 
```
out_folder="/home/phablet/$(basename "$folder")"
```

With all of that said there is nothing wrong with your solution and feel free to continue using it.

If you can test with my solution and confirm it works I will fix it the code and close this.

**Comment by RedXXIII** [comment]
My turn to apologise for my late reply! I have tried your solution, but it does not work. I removed the “/” from lines 52 and 69, but the output is to create, for example, a “.cache” folder inside the real “.cache” folder.

For now the only way to have it working is modifying the lines 58 and 75 like explained before (and lines 52 and 69 are not modified, that is, they still have the “/”). What do you think we should do?

**Comment by Intrinsically-Sublime** [comment]
Well I see no reason not to implement your fix. I have created a branch that makes the out_folder a global value of /home/phablet/ 

Can you please test it and if it works I can merge it into the main branch.
https://github.com/Intrinsically-Sublime/UT-Backup-Restore-Scripts/tree/Nesting-Output-Restore-Fix

Thanks

**Comment by RedXXIII** [comment]
I tried it and it works! Just a question: is the question Would you like to restore /home/phablet/*/ OK? If someone chooses yes, it will restore all the folders without asking for each one, right?

**Comment by Intrinsically-Sublime** [comment]
I have not been presented with the wildcard option, but after trying all my old backups it seems that if you do not have any hidden directories in the backup it presented me with.
```
Would you like to restore /home/phablet/.*[A-Za-z]/
```
which is from line 68 because and is presented because I didn't have any checks to make sure there is something to restore.

I have added checks and made sure when you skip a directory it tells you which you have skipped. All the changes are in the branch above.

**Comment by RedXXIII** [comment]
OK thanks, do you need some tests or is all OK now?

**Comment by Intrinsically-Sublime** [comment]
Testing it would be appreciated. I do not want to push it to the main branch if there is another issue. I tested it without a phone attached.

**Comment by RedXXIII** [comment]
Tested and all went good for me! Just another thing not related: in the backup script, if you want to back up something that does not exist, the terminal displays ' does not existte object '/home/phablet/folder/. That should be corrected by ' does not exist object '/home/phablet/folder/ or object /home/phablet/folder/ does not exist. What do you think?

**Comment by Intrinsically-Sublime** [comment]
I would agree the phrasing sounds wrong but I do not think it is from the script. I think it comes from adb pull command not finding something. But with that said it only presents you with folders it has found using
```
folder_array_1=($(adb shell ls -A -d /home/phablet/*/))				#create array of folders contained in the devices home folder
folder_array_2=($(adb shell ls -A -d /home/phablet/.*[A-Za-z]/))	#create array of hidden folders contained in the devices home folder
```
which means adb has found the folder. So I am not sure how it is trying to backup something that does not exist.

Can you provide the full output of running the backup script?

**Comment by RedXXIII** [comment]
hello and sorry for the delay! With my FP3+ in recovery mode I get this:

bash UT_Backup_Tool.sh

Backing up Ubuntu Touch home folders

It is recommended to skip the .cache folder to save time

ls: /home/phablet/*/: No such file or directory
ls: /home/phablet/.*[A-Za-z]/: No such file or directory
Please make sure your device is in Developer mode and connected via USB before proceeding
Would you like to start backing up your devices home folder? [Y/N] 


Goodbye

but I can see it with the command adb devices
List of devices attached
A20AAFF50202	recovery

I will try again after a normal boot.

**Comment by RedXXIII** [comment]
In recovery mode I can also have this:
bash UT_Backup_Tool.sh

Backing up Ubuntu Touch home folders

It is recommended to skip the .cache folder to save time

ls: /home/phablet/*/: No such file or directory
ls: /home/phablet/.*[A-Za-z]/: No such file or directory
Please make sure your device is in Developer mode and connected via USB before proceeding
Would you like to start backing up your devices home folder? [Y/N] y

Device backup starting.............................

Creating ./Backups/UT_Backup_22.11.25/

UT_Backup_22.11.25 complete
Run UT_Restore_Tool.sh and follow the prompts to restore the folders

But the folder is empty...

**Comment by RedXXIII** [comment]
It works with my phone on after normal reboot. Backup running...
